### ISSUE TITLE
`revolve` entire 360°

### ISSUE THREAD

**Issue by asprionj** [CadQuery/cadquery#259]
(I'm on 2.0RC1.) I want to create a simple torus, i.e. revolving a circle. If I revolve for anything shy of 360°, it works:
```
cq.Workplane('XY').circle(10).revolve(350, [-30,0,0], [-30,1,0])
```
![image](https://user-images.githubusercontent.com/13935112/71550248-d3660580-29cb-11ea-9149-8179b07905a5.png)

However, if I want the full 360°, I somehow do not get a solid (?), at least it is shown very differently:
```
cq.Workplane('XY').circle(10).revolve(360, [-30,0,0], [-30,1,0])
```
![image](https://user-images.githubusercontent.com/13935112/71550271-79b20b00-29cc-11ea-82c2-afef3316cf60.png)

What happened here? What's the difference? Is it intended?

**Comment by adam-urbanczyk** [comment]
Confirmed, it is a bug. Funny enough the following works as expected:
```python
res = cq.Workplane('XY').rect(2,2).revolve(360, (-30,0,0), (-30,1,0))
```

**Comment by adam-urbanczyk** [comment]
I found the following workaround:
```python
res = cq.Workplane('XY').circle(10).revolve(360, [-30,0,0], [-30,1,0],clean=False)
```

![afbeelding](https://user-images.githubusercontent.com/13981538/71550647-c2ba8d00-29d5-11ea-9fad-de48e23ef089.png)

**Comment by RubenRubens** [comment]
The bug is at exporting the geometry as STL. When it's exported as STEP it works fine.

**STL**
```python
import cadquery as cq
res = cq.Workplane('XY').circle(10).revolve(360, [-30,0], [-30,1], clean=True)
res.val().exportStl('test.stl', precision=0.1)
```

**STEP**
```python
import cadquery as cq
res = cq.Workplane('XY').circle(10).revolve(360, [-30,0], [-30,1], clean=True)
res.val().exportStep('test.step')
```
![bug_259](https://user-images.githubusercontent.com/51750948/73860243-8cc1c180-483b-11ea-9a92-80090ae00393.png)

**Comment by adam-urbanczyk** [comment]
@RubenRubens  I don't that this is the case:
```python
res = cq.Workplane('XY').circle(10).revolve(360, [-30,0], [-30,1], clean=True).val().isValid()
```
results in a `False`. It has nothing to do with the export. It is rather interesting that the STEP exporter does not mind.

**Comment by RubenRubens** [comment]
Good to know. 👍

**Comment by Shimwell** [comment]
I also stumbled across this bug today

A list of points connecting with a spline and rotated to 360 degrees doesn't export anything to the STL file but the STEP export worked fine.

The same list of points connected with a polyline exported fine to both STL and STEP. 

The same list of points connected with a mixture of spines and straight lines exported fine to both STL and STEP.

**Comment by jmwright** [comment]
Just tested the code posted by @Shimwell on #322 with the OCP branch (and OCCT 7.4) and the stl export worked fine. I'm hoping to get some time to put a CQ 2.0 release together within the next few days. Once that has been published, we can merge the OCP branch and move to OCCT 7.4 for the CQ 2.1 release cycle. Numerous issues will be solved by moving to OCCT 7.4 in addition to this one.

![Screenshot from 2020-04-16 08-36-59](https://user-images.githubusercontent.com/1015439/79457233-e3771480-7fbd-11ea-9112-ed36761d0db8.png)

**Comment by adam-urbanczyk** [comment]
OCP is merged into master and the full rotation works as expected:
![obraz](https://user-images.githubusercontent.com/13981538/83724289-4ee57600-a640-11ea-8ede-e9aeba7575d7.png)
